core/tests/protractor_utils/TopicsAndSkillsDashboardPage.js

```diff
@@ -315,18 +374,16 @@ var TopicsAndSkillsDashboardPage = function() {
   };
 
   this.editTopic = async function(topicName) {
-    await waitFor.elementToBeClickable(
-      topicsTabButton, 'Unable to click on topics tab.');
-    var topicElements = await _getTopicElements(topicName);
-    if (topicElements.length === 0) {
-      throw new Error('Could not find topic tile with name ' + topicName);
-    }
-    await waitFor.elementToBeClickable(
-      topicElements[0], 'Unable to click on topic: ' + topicName);
-    await topicElements[0].click();
-    await waitFor.pageToFullyLoad();
-    await waitFor.invisibilityOf(
-      topicElements[0], 'Topic editor page takes too long to load.');
+    await waitFor.visibilityOf(topicsTable,
+      'Topic table taking too long to appear.');
+    var topics = await topicNames;
+    topics.forEach((topic, index) => {
```

Comment:
You don't need to `await` ElementArrayFinders. Also, `.forEach` is unreliable with async-await

Reply:
Done.